src/StatusQ/Popups/StatusModal.qml

```diff
@@ -0,0 +1,72 @@
+import QtQuick 2.14
+import QtQuick.Controls 2.14 as QC
+
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Controls 0.1
+
+
+import "statusModal" as Spares
+
+QC.Popup {
+    id: statusModal
+
+    property Component content
+
+    property alias showBack: footerImpl.showBack
+
+    property StatusModalHeaderSettings header: StatusModalHeaderSettings {}
+    property alias buttons: footerImpl.buttons
+
+    signal backClicked()
+    signal butttonClicked(var buttonIndex)
+    signal detailsClicked()
+    signal editClicked()
```

Comment:
In other StatusQ components we've used the `*ButtonClicked()` convention so far. Maybe we can use that here as well?

Reply:
sorry, dont understand.

Reply:
So for example instead of calling it `editClicked` call it `editButtonClicked`. 

We might not need `buttonClicked` if we keep the buttons fully customizable, because then users can apply click handlers on each button themselves

Reply:
See: https://github.com/status-im/StatusQ/pull/145#discussion_r650936165